bottlerocket-settings-models/modeled-types/src/shared.rs

```diff
@@ -945,50 +945,125 @@ string_impls_for!(Lockdown, "Lockdown");
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// ApiclientCommand represents a valid Bootstrap Command. It stores the command as a vector of
+/// strings and ensures that the first argument is apiclient.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Default, Serialize)]
+pub struct ApiclientCommand(Vec<String>);
+
+impl ApiclientCommand {
+    pub fn get_command_and_args(&self) -> (&str, &[String]) {
+        self.0
+            .split_first()
+            .map(|(command, rest)| (command.as_str(), rest))
+            .unwrap_or_default()
+    }
+}
+
+// Custom deserializer added to enforce rules to make sure the command is valid.
+impl<'de> serde::Deserialize<'de> for ApiclientCommand {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let original: Vec<String> = serde::Deserialize::deserialize(deserializer)?;
+        Self::try_from(original).map_err(|e| {
+            <D::Error as serde::de::Error>::custom(format!(
+                "Unable to deserialize into ApiclientCommand: {}",
+                e
+            ))
+        })
+    }
+}
+
+impl TryFrom<Vec<String>> for ApiclientCommand {
+    type Error = error::Error;
+
+    fn try_from(input: Vec<String>) -> std::result::Result<Self, error::Error> {
+        let first_word = input.first().map(String::as_str);
+        ensure!(
+            matches!(first_word, Some("apiclient")),
```

Comment:
We probably should also confirm the second word isn't `exec` because we know that won't work for bootstrap commands since the host containers won't be up.

Reply:
I have updated it. But I am not sure if it should be here. On one hand, it **IS** an API client command, but on the other hand, if allowed, it will result in a runtime failure. The correctness of a command is not really covered here. Should we let the failure happen during the configuration itself, or should we let it error out while it is running?

Reply:
As discussed, I will be adding details about this to bootstrap-commands README in the other PR. The intention will be to make users aware of the fact that valid apiclient commands that run in a session might fail here due to dependency on services that are yet to start until that point. The restriction itself won't be added here to keep it as general as possible.

Reply:
I have added the required information in bootstrap-commands binary in core-kit PR. It can be seen in the README.md as well.